engine/engine/src/engine.cpp

```diff
@@ -1016,6 +1022,15 @@ namespace dmEngine
         // Variables need to be declared up here due to the goto's
         bool has_host_mount = dmSys::GetEnv("DM_MOUNT_HOST") != 0;
 
+        engine->m_ResourceTypeContexts.Put(dmHashString64("goc"), engine->m_Register);
+        engine->m_ResourceTypeContexts.Put(dmHashString64("collectionc"), engine->m_Register);
+        engine->m_ResourceTypeContexts.Put(dmHashString64("scriptc"), engine->m_GOScriptContext);
+        engine->m_ResourceTypeContexts.Put(dmHashString64("luac"), &engine->m_ModuleContext);
+
+        fact_result = dmResource::RegisterTypes(engine->m_Factory, &engine->m_ResourceTypeContexts);
```

Comment:
Here we make sure the resource type extensions register themselves to the factory